spec/controllers/graphql_controller_spec.rb

```diff
@@ -0,0 +1,61 @@
+RSpec.describe GraphqlController do
+
+  describe 'Authentication' do
+    it 'should return a 400 if the authentication header is missing' do
```

Comment:
This is a bit personal taste, but I'm not a fan of the `it should` naming. I think it's better to describes as facts, e.g. `it returns a 404 if the auth header is missing`. See https://rspec.rubystyle.guide/.